sap-spring-properties/src/main/java/com/sap/cloud/environment/servicebinding/SapSpringPropertiesServiceBindingAccessor.java

```diff
@@ -0,0 +1,98 @@
+import javax.annotation.Nullable;
+import java.util.*;
+
+/**
+ * A {@link ServiceBindingAccessor} that is able to load {@link SapServiceBindingsProperties}s from Spring's application
+ * properties.
+ */
+public class SapSpringPropertiesServiceBindingAccessor implements ServiceBindingAccessor
+{
+    @Nonnull
+    private static final Logger logger = LoggerFactory.getLogger(SapSpringPropertiesServiceBindingAccessor.class);
+
+    @Nonnull
+    @Override
+    public List<ServiceBinding> getServiceBindings()
+        throws ServiceBindingAccessException
+    {
+        List<ServiceBinding> serviceBindings = fetchServiceBindings();
+
+        serviceBindings
+            .forEach(binding -> logger.debug("Service binding {} found in properties.", binding.getName().get()) //NOSONAR
+            );
+
+        return serviceBindings;
+    }
+
+    @Nonnull
+    private List<ServiceBinding> fetchServiceBindings()
+    {
+        Map<String, ServiceBindingProperties> serviceBindingsProperties =
+            SapServiceBindingsPropertiesAccessor.getServiceBindingsProperties();
+
+        if( serviceBindingsProperties.isEmpty() ) {
+            logger.debug("No service bindings provided by application properties.");
+        }
+
+        return serviceBindingsProperties
+            .entrySet()
+            .stream()
+            .map(entry -> toServiceBinding(entry.getKey(), entry.getValue()))
+            .filter(Objects::nonNull)
+            .toList();
+    }
+
+    @Nullable
+    private ServiceBinding toServiceBinding(
+        @Nonnull String serviceBindingName,
+        @Nonnull final ServiceBindingProperties serviceBindingProperties )
+    {
+        if( validateServiceBindingProperties(serviceBindingProperties, serviceBindingName) ) {
+            return DefaultServiceBinding
+                .builder()
+                .copy(Collections.emptyMap())
+                .withName(
+                    StringUtils.hasText(serviceBindingProperties.getName())
+                        ? serviceBindingProperties.getName()
+                        : serviceBindingName)
+                .withServiceName(serviceBindingProperties.getServiceName())
+                .withServicePlan(serviceBindingProperties.getPlan())
+                .withTags(Arrays.asList(serviceBindingProperties.getTags()))
+                .withCredentials(serviceBindingProperties.getCredentials())
+                .build();
+        } else {
+            return null;
+        }
+    }
+
+    private boolean validateServiceBindingProperties(
+        @Nonnull final ServiceBindingProperties serviceBindingProperties,
+        @Nonnull final String serviceBindingName )
+    {
+        boolean isValid = true;
+        if( serviceBindingProperties.getServiceName() == null || serviceBindingProperties.getServiceName().isEmpty() ) {
+            logger.error("Service binding properties of {} with no service name detected.", serviceBindingName);
+            isValid = false;
```

Comment:
(Minor/Nitpick)

You could've also `return` early.
But I would consider this _authors choice_. We don't have a rule for that.

Reply:
I intentionally didn't. Validation failures indicate errors at the application's configuration and I would like to list all at once, in case there are more.